website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,244 @@
+                    ] = int(pricecomponent_value)
+
+        # add attributes to the configurator
+        chd_dict['attributes'] = str(all_attributes)
+        # dictionary is complete, create configurator
+        try:
+            new_chd = request.env['chd.product_configurator'].create(chd_dict)
+        except orm.except_orm as e:
+            errormsg = e.value
+        except AssertionError as a:
+            errormsg = a.value
+        # add accessories and price components selections to the configurator
+        list = []
+        for key in form_data:
+            # get only accessories that have been checked
+            if ('accessoryid_' in key) and form_data[key] != 0:
+                # extrapolate the id encoded the name
+                accessory_id = int(key.split('_')[1])
+                # get the associated value by choosing the field with the
+                # right name accessoryid_{id}=on/off and the associated
+                # quantity would be qtyaccessoryid_{id}=9898
+                accessory_qty = int(form_data[key])
+                list.append((0, 0, {
+                    'product_id': accessory_id,
+                    'configurator_id': new_chd.id,
+                    'quantity': accessory_qty, }
+                    ))
+        if list != []:
+            new_chd.write({'accessoire_line_ids':list})
+        # oduct configurator is ready, we can now calculate options
+        # model refers to old API model, self.pool
+        # is not available in controller context (praise the lord for Holger!)
+        try:
+            res = new_chd._model.calculate_price(
+                request.cr, request.uid, [new_chd.id], context=request.context)
+            results = chd_results_model.search([
+                ('configurator_id', '=', new_chd.id)])
+        except orm.except_orm as e:
+                errormsg += e.value
+
+        if errormsg != "":
+            product_type_model = request.env['product.type']
+            product_product_model = request.env['product.product']
+            curr_types = product_type_model.search([
+                ('product_option_ids', 'in', [int(form_data['product_id'])])
+                ])
+            avail_accessories = product_product_model.search(
+                [('id', 'in', curr_product_id.chd_accessoire_ids.ids)])
+            curr_chd_price_component_ats = pc_at_model.search(
+                [
+                    ('id', 'in', curr_product_id.attribute_template_ids.ids),
+                    ('active', '=', True)
+                    ])
+            return request.render(
+                'website_chd_product_configurator.configurator', {
+                    'curr_product_id': curr_product_id,
+                    'curr_types': curr_types,
+                    'avail_accessories': avail_accessories,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'errormsg': errormsg,
```

Comment:
what happens if you get multiple errors above? Given you just concat them, you'll have a badly readly readable string here ('Error1: this is wrongError2: this is wrong too'). Wouldn't if be better to make `errormsg` a list and append all messages there?